common/src/main/java/org/apache/falcon/metadata/InstanceRelationshipGraphBuilder.java

```diff
@@ -256,34 +256,47 @@ public void addEvictedInstance(WorkflowExecutionContext context) throws FalconEx
         }
     }
 
-
     public void addImportedInstance(WorkflowExecutionContext context) throws FalconException {
+        addImportExportInstanceHelper(context, RelationshipLabel.DATASOURCE_IMPORT_EDGE);
+    }
 
-        String feedName = context.getOutputFeedNames();
-        String feedInstanceDataPath = context.getOutputFeedInstancePaths();
+    public void addExportedInstance(WorkflowExecutionContext context) throws FalconException {
+        addImportExportInstanceHelper(context, RelationshipLabel.DATASOURCE_EXPORT_EDGE);
+    }
+
+    private void addImportExportInstanceHelper(WorkflowExecutionContext context,
+                                               RelationshipLabel label) throws FalconException {
+        String feedName = (label == RelationshipLabel.DATASOURCE_IMPORT_EDGE)
+                ? context.getOutputFeedNames() : context.getInputFeedNames();
+        String feedInstanceDataPath = (label == RelationshipLabel.DATASOURCE_IMPORT_EDGE)
+                ? context.getOutputFeedInstancePaths() : context.getInputFeedInstancePaths();
         String datasourceName = context.getDatasourceName();
         String sourceClusterName = context.getSrcClusterName();
```

Comment:
Nope. why? The context has the datasource and cluster from which data is exported out. Also, same for the IMPORT case. 

BTW, while all the comments appear in the latest version, this one appears on the older version (on the left side) :-)